Added Curl to Conan Packages

lancejennings opened the pull request:
Added the Curl library for conan packages

lancejennings committed:
Added Curl to Conan Packages

Added the Curl library for conan packages

DavidZemon commented on curl/test_package/conanfile.py:
```diff
@@ -0,0 +1,21 @@
+        cmake.build()
+
+    def test(self):
+        if tools.cross_building(self.settings):
```
This looks like a copy/paste mistake. No reason cross-compiling curl shouldn't work. Can you check some other test_package Conan recipes and compare?

lancejennings replied:
Most other packages do not have the "if tools.cross_building(self.settings):" check, perhaps it is best if we leave this part out?

DavidZemon replied:
Yep, agreed

DavidZemon commented on curl/test_package/CMakeLists.txt:
```diff
@@ -0,0 +1,13 @@
+project(CurlTest C)
+
+include("${CMAKE_BINARY_DIR}/conanbuildinfo.cmake")
+conan_basic_setup(TARGETS)
```
So, believe it or not, you actually don't have to use the Conan way of linking against Curl. CMake provides a FindCURL module: https://cmake.org/cmake/help/v3.0/module/FindCURL.html

DavidZemon replied:
Take a look at libxml2, it shows a better example for that

lancejennings committed:
Removed cross_building option check since we do not need it for this library

DavidZemon commented on libtermcap/test_package/conanfile.py:
```diff
@@ -11,10 +11,7 @@ def build(self):
-            self.output.info('This test always fails when cross-compiling. Skipping test.')
-        else:
-            self.run('ctest --output-on-failure')
+        self.run('ctest --output-on-failure')
```
Did you really intend to change this file?

lancejennings replied:
No. I got confused since every folder has a test_package directory with a conanfile.py in it. I needed to collapse some of those directories in CLion so I wouldn't get confused with what I was actually modifying.

lancejennings committed:
Used the find_package for curl instead of using conan

Since curl provides a find curl module, we can use find_package to populate the include and library link path variables for the cmake recipe.

lancejennings committed:
Readded cross building check for libtermcap and removed for curl

DavidZemon commented on curl/test_package/CurlTest.c:
```diff
@@ -0,0 +1,56 @@
+#include <curl/curl.h>
+
+/*
+ * This test example file was copied from https://curl.haxx.se/libcurl/c/https.html
```
Exactly the right way to do it.

DavidZemon commented on curl/test_package/conanfile.py:
```diff
@@ -0,0 +1,18 @@
+        cmake.build()
+
+    def test(self):
+            self.run('ctest --output-on-failure')
```
Indentation.

lancejennings committed:
Fixed indentation for test() in conanfile.py

DavidZemon commented on curl/conanfile.py:
```diff
@@ -0,0 +1,85 @@
+        if self.options.with_zlib:
+            self.requires("zlib/1.2.11@conan/stable")
+        if self.options.with_ssl:
+            self.requires("OpenSSL/1.0.2n@conan/stable")
```
Can you switch this to 1.0.2o? Without that change, you'll run into the same problem here that we're seeing with mosquitto: https://ci.redlion.net/viewLog.html?buildId=2390&buildTypeId=MqttMosquitto_Conan&tab=buildLog&_focus=458
And for some reason, upstream conan guys fixed it for OpenSSL v1.0.2o :) (I tested it on my machine just now)

lancejennings committed:
Changed OpenSSL version from 1.0.2n to 1.0.2o
